### PR TITLE
Pytest Style analysis/test_psa.py

### PR DESCRIPTION
**Pull request by utkbansal**
Fixes #

Changes made in this Pull Request:
 - 

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

File: testsuite/MDAnalysisTests/analysis/test_psa.py [hunk 1]
```diff
@@ -24,71 +24,78 @@
...
+    def test_reversal_frechet(self, frech_matrix):
         err_msg = "Frechet distances did not increase after path reversal"
-        assert_(self.frech_matrix[1,2] >= self.frech_matrix[0,1], err_msg)
+        assert frech_matrix[1,2] >= frech_matrix[0,1], err_msg
 
-    def test_dendrogram_produced(self):
+    def test_dendrogram_produced(self, plot_data):
         err_msg = "Dendrogram dictionary object was not produced"
-        assert_(type(self.plot_data[1]) is dict, err_msg)
+        assert type(plot_data[1]) is dict, err_msg
```
Comment (jbarnoud) [inline testsuite/MDAnalysisTests/analysis/test_psa.py]: `isinstance` is probably more appropriate than `type(...) is ...`.

File: testsuite/MDAnalysisTests/analysis/test_psa.py [hunk 2]
```diff
@@ -176,152 +174,163 @@ def test_dist_mat_to_vec_func_bad_integers(self):
...
+                np.mean(np.amin(distance_matrix, axis=1))) / 2.0
 
-    def setUp(self):
-        super(TestWeightedAvgHausdorffSymmetric, self).setUp()
-        self.h = PSA.hausdorff_wavg
-        self.distance_matrix = scipy.spatial.distance.cdist(self.path_1,
-                                                            self.path_2)
-        self.expected = (np.mean(np.amin(self.distance_matrix, axis=0)) +
-                         np.mean(np.amin(self.distance_matrix, axis = 1))) / 2.
+    # TODO: Fix this hack! without the fixture the test says I passed 3
```
Comment (jbarnoud) [inline testsuite/MDAnalysisTests/analysis/test_psa.py]: Could `h` be a class attribute rather than a fixture?

Comment (utkbansal) [inline testsuite/MDAnalysisTests/analysis/test_psa.py]: I did try that, but when I make it a class attribute, the test fails saying that `h` expected 2 parameters but got 3.

Comment (jbarnoud) [inline testsuite/MDAnalysisTests/analysis/test_psa.py]: It looks indeed like `h`, when defined as a class attribute, is taken as a method. I guess you could define it as `h = staticmethod(PSA.hausdorff_wavg)` but I am not sure it is much better than the fixture. Maybe somebody will come with an idea. If not, we'll survive with the fixture.

**Pull request merged**